src/layout/sidepanel.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from "react";
+import ReactDOM from "react-dom/client";
+
+import { TabName } from "../types";
+
+import Header from "../layout/Header";
+import Body from "./Body";
+import Footer from "./Footer";
+import ChatBox from "../components/chatBox/ChatBox";
+import Content from "../components/content/Content";
+import Interest from "../components/interest/Interest";
+import "../styles/style.scss";
+
+const Sidepanel = () => {
+  const [activeTab, setActiveTab] = useState<TabName>("ChatBox");
+
+  const componentMap: Record<TabName, JSX.Element> = {
+    ChatBox: <ChatBox />,
+    Content: <Content />,
+    Interest: <Interest />,
+  };
+
+  useEffect(() => {
+    chrome.runtime.onMessage.addListener((message) => {
+      if (
+        message.type === "SET_ACTIVE_TAB" &&
+        message.payload in componentMap
+      ) {
+        setActiveTab(message.payload as TabName);
+      }
+    });
+
+    chrome.runtime.sendMessage({ type: "GET_ACTIVE_TAB" }, (response) => {
+      if (response?.activeTab && response.activeTab in componentMap) {
+        setActiveTab(response.activeTab as TabName);
```

Comment:
Just one thing, why both SET_ACTIVE_TAB and GET_ACTIVE_TAB triggers setActiveTab hook

Reply:
Get_active_tab is basically for other components comunnication about the active tab status, since we removed that logic later pr with redux centralized state management we replace that logic with simpler but effective one.